src/pages/VenuePage/VenuePage.tsx

```diff
@@ -38,25 +42,105 @@ import { useVenueId } from "hooks/useVenueId";
 // import { useVenueAccess } from "hooks/useVenueAccess";
 import useConnectCurrentVenue from "hooks/useConnectCurrentVenue";
 
-import { CountDown } from "components/molecules/CountDown";
-import { LoadingPage } from "components/molecules/LoadingPage/LoadingPage";
-// import { AccessDeniedModal } from "components/atoms/AccessDeniedModal/AccessDeniedModal";
-import TemplateWrapper from "./TemplateWrapper";
-
 import { updateTheme } from "./helpers";
 
+import Login from "pages/Account/Login";
+
 import "./VenuePage.scss";
 
-import Login from "pages/Account/Login";
-import { useRelatedVenues } from "hooks/useRelatedVenues";
+const CountDown = React.lazy(() =>
+  import("components/molecules/CountDown").then((m) => ({
+    default: m.CountDown,
+  }))
+);
+const LoadingPage = React.lazy(() =>
+  import("components/molecules/LoadingPage/LoadingPage").then((m) => ({
+    default: m.LoadingPage,
+  }))
+);
+const TemplateWrapper = React.lazy(() => import("./TemplateWrapper"));
 
 // @debt Refactor this constant into settings, or types/templates, or similar?
 const hasPaidEvents = (template: VenueTemplate) => {
   return template === VenueTemplate.jazzbar;
 };
 
+interface PreloadProps {
+  venue?: AnyVenue;
+}
+
+const Preload: React.FC<PreloadProps> = ({ venue }) => (
+  <Helmet>
+    {
+      // preload the icon only when TemplateWrapper with the navbar icon aren't loaded lazily
+      // <link href={SPARKLE_ICON} rel="preload" as="image" />
+    }
+    {venue?.mapBackgroundImageUrl && (
+      <link href={venue?.mapBackgroundImageUrl} rel="preload" as="image" />
+    )}
+    {
+      // NOTE: if there is significant number of rooms, preload might degrade performance, cut down to few
+      (venue?.rooms ?? []).map(
+        (room) =>
+          room?.image_url && (
+            <link
+              key={room?.image_url}
+              href={room.image_url}
+              rel="preload"
+              as="image"
+            />
+          )
+      )
```

Comment:
If this stays it would need to be `useMemo`'d or similar.

Reply:
Superseded by e9da1d5788e915f6de13a7aa0315601017c75b73 .

Hook named `usePreloadAssets` with `useEffect` inside dealing with `document.head` using plain DOM functions. Not sure if/how `useMemo` will remain relevant.

Reply:
Can use confirmation if no more changes are needed before closing this discussion

Reply:
Sounds fine. If I have any more specific comments i'll make them inline in a new thread on next review.